### PR TITLE
fix: group/events API raises 404 if group is deleted

### LINKED ISSUE #7538: Implement Organizer API with list of EventIDs of Organizers
Here is a question of a user about REST APIs which we also need for the organizer feature:

Is it possible to use this REST interface to query all upcoming events of an organizer? Specifically, that means, can I query all upcoming published events of a user with a web service request and receive the EventIDs? I can then use the EventIDs for the individual queries of the specific event data.

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #7538 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal Review

File: app/api/events.py [hunk 1]
```diff
@@ -349,6 +355,11 @@ def query(self, view_kwargs):
             )
 
         if view_kwargs.get('group_id') and 'GET' in request.method:
+            group = safe_query_without_soft_deleted_entries(
```
Comment (iamareebjamal) [inline app/api/events.py:358]: Just use safe_query

File: app/api/events.py [hunk 2]
```diff
@@ -349,6 +355,11 @@ def query(self, view_kwargs):
...
+                Group, 'id', view_kwargs.get('group_id'), 'group_id'
+            )
+            if not group:
+                raise NotFoundError({'source': ''}, 'Group id doesn\'\t exist')
```
Comment (iamareebjamal) [inline app/api/events.py:362]: No need, safe query automatically raises 404 if object is not found

File: app/api/events.py (hunk 1 above)
Comment (codedsun) [reply app/api/events.py]: Will it filter soft deleted enteries?

**Comment by codedsun** [comment]
@iamareebjamal - Review! Done changes

**Pull request merged**